src/main/java/net/dv8tion/jda/api/hooks/AnnotatedEventManager.java

```diff
@@ -114,32 +118,31 @@ private void updateMethods()
         methods.clear();
         for (Object listener : listeners)
         {
-            boolean isClass = listener instanceof Class;
-            Class<?> c = isClass ? (Class) listener : listener.getClass();
-            Method[] allMethods = c.getDeclaredMethods();
-            for (Method m : allMethods)
-            {
-                if (!m.isAnnotationPresent(SubscribeEvent.class) || (isClass && !Modifier.isStatic(m.getModifiers())))
-                {
-                    continue;
-                }
-                Class<?>[] pType  = m.getParameterTypes();
-                if (pType.length == 1 && GenericEvent.class.isAssignableFrom(pType[0]))
-                {
-                    Class<?> eventClass = pType[0];
-                    if (!methods.containsKey(eventClass))
-                    {
-                        methods.put(eventClass, new ConcurrentHashMap<>());
-                    }
+            registerListenerMethods(listener);
+        }
+    }
 
-                    if (!methods.get(eventClass).containsKey(listener))
-                    {
-                        methods.get(eventClass).put(listener, new CopyOnWriteArrayList<>());
-                    }
+    private void registerListenerMethods(Object listener)
+    {
+        boolean isClass = listener instanceof Class;
+        Class<?> c = isClass ? (Class<?>) listener : listener.getClass();
+        Method[] allMethods = c.getDeclaredMethods();
+        for (Method m : allMethods)
+        {
+            if (!m.isAnnotationPresent(SubscribeEvent.class))
+                continue;
+            //Skip member methods if listener is a Class
+            if (isClass && !Modifier.isStatic(m.getModifiers()))
+                continue;
 
-                    methods.get(eventClass).get(listener).add(m);
-                }
-            }
+            final Class<?>[] parameterTypes = m.getParameterTypes();
+            if (parameterTypes.length != 1 || !GenericEvent.class.isAssignableFrom(parameterTypes[0]))
+                throw new IllegalArgumentException("Method '" + m + "' must have at most 1 parameter, which implements GenericEvent");
```

Comment:
I think its better to log this as warning instead of throwing an error

Reply:
That's not necessarily true, sometimes people might use this for their own internal usage. For instance, you might have a custom event system making use of the same annotation. Since we never previously required this, I see no reason to introduce this limitation now.